### PR TITLE
Add secure boot code

### PR DESCRIPTION
**Pull request by mjg59**
Add shim and update grub to incorporate required functionality

### CODE REVIEW

File: sys-boot/shim/shim-9999.ebuild [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	default_src_unpack
+}
+
+src_compile() {
+	make EFI_PATH=/usr/lib shim.efi
+}
+
+src_install() {
+	insinto /usr/lib/shim
+	newins "shim.efi" shim.efi
```
Comment (marineam) [inline sys-boot/shim/shim-9999.ebuild]: When the name doesn't change you can use doins

File: sys-boot/shim/shim-9999.ebuild [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+DESCRIPTION="UEFI Shim loader"
+HOMEPAGE="https://github.com/rhinstaller/shim"
+
+LICENSE="BSD"
+SLOT="0"
+IUSE=""
+
+RDEPEND=""
+DEPEND="sys-boot/gnu-efi"
```
Comment (marineam) [inline sys-boot/shim/shim-9999.ebuild]: Doesn't shim depend on some other stuff to build like sb-sign or... I forgot

Comment (mjg59) [inline sys-boot/shim/shim-9999.ebuild]: Ah, yeah, it depends on openssl. I'll add that.

File: sys-boot/shim/shim-9999.ebuild [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# Copyright 1999-2014 Gentoo Foundation
```
Comment (marineam) [inline sys-boot/shim/shim-9999.ebuild]: Yeah, vim does this by default but we are CoreOS, Inc. :)

**Comment by marineam** [comment]
one last header nit and then LGTM

**Pull request merged**